bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java

```diff
@@ -318,74 +325,77 @@ public void initialize() {
         if (configError != null) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, configError);
         } else {
-            // Set custom input labels
-            try {
-                if (config.inputLabelCd != null && !config.inputLabelCd.isEmpty()) {
-                    rotelModel.setSourceLabel("CD", config.inputLabelCd);
-                }
-                if (config.inputLabelTuner != null && !config.inputLabelTuner.isEmpty()) {
-                    rotelModel.setSourceLabel("TUNER", config.inputLabelTuner);
-                }
-                if (config.inputLabelTape != null && !config.inputLabelTape.isEmpty()) {
-                    rotelModel.setSourceLabel("TAPE", config.inputLabelTape);
-                }
-                if (config.inputLabelPhono != null && !config.inputLabelPhono.isEmpty()) {
-                    rotelModel.setSourceLabel("PHONO", config.inputLabelPhono);
-                }
-                if (config.inputLabelVideo1 != null && !config.inputLabelVideo1.isEmpty()) {
-                    rotelModel.setSourceLabel("VIDEO1", config.inputLabelVideo1);
-                }
-                if (config.inputLabelVideo2 != null && !config.inputLabelVideo2.isEmpty()) {
-                    rotelModel.setSourceLabel("VIDEO2", config.inputLabelVideo2);
-                }
-                if (config.inputLabelVideo3 != null && !config.inputLabelVideo3.isEmpty()) {
-                    rotelModel.setSourceLabel("VIDEO3", config.inputLabelVideo3);
-                }
-                if (config.inputLabelVideo4 != null && !config.inputLabelVideo4.isEmpty()) {
-                    rotelModel.setSourceLabel("VIDEO4", config.inputLabelVideo4);
-                }
-                if (config.inputLabelVideo5 != null && !config.inputLabelVideo5.isEmpty()) {
-                    rotelModel.setSourceLabel("VIDEO5", config.inputLabelVideo5);
-                }
-                if (config.inputLabelVideo6 != null && !config.inputLabelVideo6.isEmpty()) {
-                    rotelModel.setSourceLabel("VIDEO6", config.inputLabelVideo6);
-                }
-                if (config.inputLabelUsb != null && !config.inputLabelUsb.isEmpty()) {
-                    rotelModel.setSourceLabel("USB", config.inputLabelUsb);
-                }
-                if (config.inputLabelMulti != null && !config.inputLabelMulti.isEmpty()) {
-                    rotelModel.setSourceLabel("MULTI", config.inputLabelMulti);
+            for (RotelSource src : rotelModel.getSources()) {
+                // Consider custom input labels
+                if (src.getName().equals("CD") && config.inputLabelCd != null && !config.inputLabelCd.isEmpty()) {
```

Comment:
To make it a little more readable and less use of if-else statements. How about this:
```
String label = null;
switch (src.getname()) {
case "CD": 
  label = config.inputLabelCd;
  break;
...
default:
  label = src.getLabel();
}
if (label != null && !label.empty()) {
  sourcesLabels.put(src, label);
}
```

Reply:
You're right, that would be better.

Reply:
Fixed.